src/eko/evolution_operator/__init__.py

```diff
@@ -182,7 +192,16 @@ def quad_ker(
 
     # compute the actual evolution kernel
     if ker_base.is_singlet:
-        gamma_singlet = ad.gamma_singlet(order, ker_base.n, nf)
+        if is_polarized:
+            if is_time_like:
+                gamma_singlet = ad_pt.gamma_singlet(order, ker_base.n, nf)
+            else:
+                gamma_singlet = ad_ps.gamma_singlet(order, ker_base.n, nf)
+        else:
+            if is_time_like:
+                gamma_singlet = ad_ut.gamma_singlet(order, ker_base.n, nf)
+            else:
+                gamma_singlet = ad_us.gamma_singlet(order, ker_base.n, nf)
```

Comment:
Here I'd be tempted to go for dynamic imports, to avoid the incredibly long (and repeated) branching. More or less as we are doing for yadism.

What do you think? @felixhekhorn

Reply:
I didn't try, but I'm quite convinced numba doesn't like dynamic imports since it needs to know ahead of time which functions are inside and need a compilation

Reply:
You're definitely right, I keep forgetting the point where Numba kicks in...